File: python-solutions/container-with-most-water.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_height = 0
        left = 0
        right = len(height) - 1
        while left < right:
            current_max = min(height[left], height[right]) * (right - left)
            max_height = max(max_height, current_max)
            if (height[left] < height[right]):
                left += 1
            else:
                right -= 1
        return max_height

File: javascript-solutions/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
	let max = 0,
		left = 0,
		right = height.length - 1;

	while (left < right) {
		const currentMax = Math.min(height[left], height[right]) * (right - left);
		max = Math.max(currentMax, max);
		height[right] > height[left] ? left++ : right--;
	}
	return max;
};
